Signal/src/environment/Migrations/OWS105AttachmentFilePaths.m

```diff
@@ -0,0 +1,58 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWS105AttachmentFilePaths.h"
+#import <SignalServiceKit/TSAttachmentStream.h>
+#import <YapDatabase/YapDatabaseTransaction.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+// Increment a similar constant for every future DBMigration
+static NSString *const OWS105AttachmentFilePathsMigrationId = @"105";
+
+@implementation OWS105AttachmentFilePaths
+
++ (NSString *)migrationId
+{
+    return OWS105AttachmentFilePathsMigrationId;
+}
+
+- (void)runUpWithTransaction:(YapDatabaseReadWriteTransaction *)transaction
+{
+    OWSAssert(transaction);
+
+    NSMutableArray<TSAttachmentStream *> *attachmentStreams = [NSMutableArray new];
+    [transaction enumerateKeysAndObjectsInCollection:TSAttachmentStream.collection
+                                          usingBlock:^(NSString *key, TSAttachment *attachment, BOOL *stop) {
+                                              if (![attachment isKindOfClass:[TSAttachmentStream class]]) {
+                                                  return;
+                                              }
+                                              TSAttachmentStream *attachmentStream = (TSAttachmentStream *)attachment;
+                                              if (attachmentStream.hasUnsavedFilePath) {
```

Comment:
Not a bad idea, although the migration doesn't need to `ensureFilePath`; (all of) the initializers have to do that.